backend/business-partner-agent/src/main/java/org/hyperledger/bpa/impl/aries/PresentationRequestHelper.java

```diff
@@ -0,0 +1,108 @@
+ * https://github.com/hyperledger-labs/business-partner-agent
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hyperledger.bpa.impl.aries;
+
+import org.hyperledger.aries.api.present_proof.*;
+import org.hyperledger.bpa.api.exception.PresentationConstructionException;
+
+import java.util.*;
+import com.google.gson.JsonObject;
+
+public class PresentationRequestHelper {
+
+    /**
+     * Returns a valid PresentationRequest if possible (no preference)
+     * 
+     * @param presentationExchange {@link PresentationExchangeRecord}
+     * @param validCredentials     {@link List<PresentationRequestCredentials>}
+     * @return {@link PresentationRequest}
+     */
+    public static Optional<PresentationRequest> buildAny(
+            PresentationExchangeRecord presentationExchange,
+            List<PresentationRequestCredentials> validCredentials)
+            throws PresentationConstructionException {
+        Optional<PresentationRequest> result = Optional.empty();
+        result = Optional.of(PresentationRequest.builder()
+                .requestedAttributes(buildRequestedAttributes(
+                        presentationExchange.getPresentationRequest().getRequestedAttributes(),
+                        validCredentials))
+                .requestedPredicates(buildRequestedPredicates(
+                        presentationExchange.getPresentationRequest().getRequestedPredicates(),
+                        validCredentials))
+                .build());
+        return result;
+
+    }
+
+    private static Map<String, PresentationRequest.IndyRequestedCredsRequestedAttr> buildRequestedAttributes(
+            Map<String, PresentProofRequest.ProofRequest.ProofAttributes> requestedAttributes,
+            List<PresentationRequestCredentials> validCredentials)
+            throws PresentationConstructionException {
+
+        Map<String, PresentationRequest.IndyRequestedCredsRequestedAttr> result = new HashMap<>();
+
+        for (Map.Entry<String, PresentProofRequest.ProofRequest.ProofAttributes> reqAttr : requestedAttributes
+                .entrySet()) {
+            List<JsonObject> restrictions = reqAttr.getValue().getRestrictions();
+            Optional<PresentationRequestCredentials> cred = validCredentials.stream()
+                    .filter(vc -> vc.getCredentialInfo().getSchemaId().equals(
+                            restrictions.get(0).get("schema_id").getAsString()))
```

Comment:
What happens if schemaId is not requested but something else?

Reply:
It will likely crash, but I wasn't focused on define a general use proof construction so this can process requests made by other BPA's but is fragile to anything that does not match that structure. 

I've added TODO's to acknowledge the assumptions and limitations of it.

Reply:
I have to think about that one as its probably better to also move this to the aries wrapper and keep it generic the same way I did with the proof handling where there is a pojo mapper for everything.

Reply:
i had to modify it to handle the holder being the initiator. i think it satisfies all basic cases.